sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ThisShouldNotBeExposedFromConstructors.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using static Renamed = Tests.Diagnostics.Other.StaticMethod;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public Program()
+        {
+            StaticMethod(null);
+            StaticMethod(this);
+            StaticMethod(((this)));
+            StaticProperty = this;
+            StaticProperty = ((this));
+
+            Other.StaticMethod(this); // Noncompliant
+            Other.StaticList.Add(this); // Noncompliant
+            Other.StaticProperty = this; // Noncompliant
+            ProgramsStatic.Add(this); // Noncompliant
+            InstanceList.Add(this); // Noncompliant
+            this.InstanceList.Add(this); // Noncompliant
+            InstanceProperty = this;

```

Comment:
Isn't it supposed to be caught?

Reply:
as far as I can understand the rule, you are not supposed to provide potentially uninitialized instance to "other" classes and instances and it is ok if you give it to the same instance...